ftw/casauth/plugin.py

```diff
@@ -104,13 +110,63 @@ def authenticateCredentials(self, credentials):
         if not userid:
             return None
 
+        result = self.login_user(userid)
+        if not result:
+            return None
+
+        return userid, userid
+
+    def login_user(self, userid):
         pas = self._getPAS()
         info = pas._verifyUser(pas.plugins, user_id=userid)
         if info is None:
             return None
+
+        mtool = getToolByName(getSite(), 'portal_membership')
+        member = mtool.getMemberById(userid)
+        if member is None:
+            return None
+
+        first_login = self.set_login_times(member)
+
+        self.fire_login_events(first_login, member)
+
+        self.expire_clipboard()
+
+        mtool.createMemberArea(member_id=userid)
+
         pas.updateCredentials(self.REQUEST, self.REQUEST.RESPONSE, userid, '')
+        return member
 
-        return userid, userid
+    def set_login_times(self, member):
+        """The return value indicates if this is the first logged
```

Comment:
A docstring makes this method publishable through the web. Make a comment instead.